src/lifecycle/properties-init.js

```diff
@@ -79,13 +79,20 @@ function createNativePropertyDefinition (name, opts) {
 
   prop.get = function () {
     const info = data(this, `api/property/${name}`);
+    const internalValue = info.internalValue;
 
     if (opts.get) {
-      return opts.get(this);
+      return opts.get(this, { name, internalValue });
     }
 
-    return info.internalValue;
+    return internalValue;
   };
+  
+  prop.init = function () {
```

Comment:
Instead of putting more logic in the `properties-ready` lifecycle, this seemed a better place as it co-locates the internal `data`.